src/sentry/auth/access.py

```diff
@@ -238,21 +238,29 @@ def from_request(request, organization=None, scopes=None):
         return from_sentry_app(request.user, organization=organization)
 
     if is_active_superuser(request):
-        # we special case superuser so that if they're a member of the org
-        # they must still follow SSO checks, but they gain global access
-        try:
-            member = OrganizationMember.objects.get(
-                user=request.user,
-                organization=organization,
-            )
-        except OrganizationMember.DoesNotExist:
-            requires_sso, sso_is_valid = False, True
-        else:
-            requires_sso, sso_is_valid = _sso_params(member)
-
         team_list = ()
-
         project_list = ()
+        requires_sso = False
+        sso_is_valid = True
+        permissions = frozenset()
```

Comment:
Creating a permission-less superuser that is also an automated system is riskier than the 'robot' user that only has the permissions (new scopes) that it needs. If there was a leak/breach the bot token would have no value on all API endpoints other than those it had the right scopes to access.

Reply:
@mattrobenolt and I had a discussion about that. The internal systems that would authenticate as such a user could also directly access any other internal system such as the database, causing much more harm. That being said, I agree that giving this unscoped super user access is just increasing the attack vector.

The alternative to this would be to either add a specific authentication class or permission class  just for the symbolicator, which is only configured on a single endpoint (debug files), but still gives super user access (since a single user needs full scope access to all organizations). We repeat this for every other internal service that we're building since we'd want to isolate their permissions in that case. This seemed a bit too "hard coded", though.

FWIW, in the current form the system token would be negotiated between the symbolicator service and Sentry, so we could cycle it any time we want. We could even swap it daily.

Reply:
Adding to the above, essentially this is a matter of preference on how we want to view this:

1. The token could stand for a service that is essentially part of Sentry, except that it runs in its own process. E.g., similar to the current native plugin, the symbolicator is acting like an externalized piece of code, except it goes through the API to query the database.

2. We could create the notion of authenticated and scoped applications that are implicitly authorized for every organization but only for one specific use case each (e.g. downloading symbols). I believe that would require a new authentication model alltogether, plus separate endpoints.

Reply:
If the system token only had the scopes it needs to do its present day work the risk would be less.

> We could create the notion of authenticated and scoped applications that are implicitly authorized for every organization but only for one specific use case each (e.g. downloading symbols). I believe that would require a new authentication model alltogether, plus separate endpoints.

Would we need a new authentication model? The current token authenticator allows us to resolve the system token, the model you've described sounds like multiple system tokens that have different scopes attached to them. You may need a different user object though, but you have something along those lines now by adding `is_system` to the superuser definition. Having system-users be separate from superusers allows us to keep system users on a tighter leash.